src/test/java/uk/gov/cslearning/record/api/CourseRecordControllerTest.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.cslearning.record.api;
+
+import org.assertj.core.util.Lists;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.cslearning.record.dto.CourseRecordDto;
+import uk.gov.cslearning.record.dto.factory.ErrorDtoFactory;
+import uk.gov.cslearning.record.service.CourseRecordService;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+@WebMvcTest({CourseRecordController.class, ErrorDtoFactory.class})
+@RunWith(SpringRunner.class)
+@WithMockUser(username = "user")
+public class CourseRecordControllerTest {
```

Comment:
- Any exception handling?
- Only positive tests?
- Any code coverage done?

Reply:
Exception handling is as per the existing error handling framework. Code coverage is checked locally. Ideally, code coverage report should be generated as part of the Jenkins build pipeline rather then local development environment which may differ on developer's machines.